tools/gfx/renderer-shared.h

```diff
@@ -718,20 +716,57 @@ class ShaderObjectBaseImpl : public ShaderObjectBase
                 SLANG_ASSERT(false && "Unexpected specialization argument kind.");
                 return SLANG_FAIL;
             }
-            m_userProvidedSpecializationArgs->add(extendedType);
+            list.add(extendedType);
         }
         return SLANG_OK;
     }
 
-    // Appends all types that are used to specialize the element type of this shader object in
-    // `args` list.
-    virtual Result collectSpecializationArgs(ExtendedShaderObjectTypeList& args) override
+    virtual SLANG_NO_THROW Result SLANG_MCALL setSpecializationArgs(
+        ShaderOffset const& offset,
+        const slang::SpecializationArg* args,
+        uint32_t count) override
     {
-        if (m_userProvidedSpecializationArgs)
+        auto layout = getLayout();
+
+        // If the shader object is a container, delegate the processing to
+        // `setSpecializationArgsForContainerElements`.
+        if (layout->getContainerType() != ShaderObjectContainerType::None)
```

Comment:
I'm still a little confused about how the property of being or not being a container works, so I'm mostly reading this as "what to do in the structured-buffer case."

Reply:
Yeah, this is just for the structured-buffer case for now.